src/components/interface/voter/voter.h

```diff
@@ -8,7 +8,7 @@ typedef enum {
 } replica_type;
 
 // to be fleshed out at a later time - long term solution is to have data in cbufs
-int nread(spdid_t spdid, replica_type from, size_t sz);
+size_t nread(spdid_t spdid, replica_type from, size_t sz);
 int nwrite(spdid_t spdid, replica_type to, size_t sz);
```

Comment:
I think the confusion might be that nwrite no longer takes in data. That's because after I changed data from an int to an actual buffer, I had it mapped into both the voter and ping/pong's address space. One write buffer for ping and one for pong. So ping/pong now just write into their buffer and only tell the voter how many bytes they have written. If the confusion is for the int return value, that's just success/fail for me. Does that help?